dd-sdk-android-fresco/src/test/kotlin/com/datadog/android/fresco/DatadogFrescoCacheListenerTest.kt

```diff
@@ -0,0 +1,185 @@
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.fresco
+
+import com.datadog.android.fresco.utils.Configurator
+import com.datadog.android.rum.GlobalRum
+import com.datadog.android.rum.RumErrorSource
+import com.datadog.android.rum.RumMonitor
+import com.datadog.tools.unit.forge.aThrowable
+import com.datadog.tools.unit.getStaticValue
+import com.facebook.cache.common.CacheEvent
+import com.facebook.imagepipeline.cache.BitmapMemoryCacheKey
+import com.nhaarman.mockitokotlin2.argumentCaptor
+import com.nhaarman.mockitokotlin2.doReturn
+import com.nhaarman.mockitokotlin2.eq
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.annotation.RegexForgery
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import java.io.IOException
+import java.util.concurrent.atomic.AtomicBoolean
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(
+        MockitoExtension::class,
+        ForgeExtension::class
+    )
+)
+@ForgeConfiguration(value = Configurator::class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+internal class DatadogFrescoCacheListenerTest {
+
+    private lateinit var underTest: DatadogFrescoCacheListener
+
+    @Mock
+    lateinit var mockRumMonitor: RumMonitor
+
+    @Mock
+    lateinit var mockCacheEvent: CacheEvent
+
+    lateinit var fakeEventKey: BitmapMemoryCacheKey
+
+    @RegexForgery(value = "http://[a-z]+\\.com")
+    lateinit var fakeCacheEventUri: String
+
+    // region Unit Tests
+
+    @BeforeEach
+    fun `set up`() {
+        GlobalRum.registerIfAbsent(mockRumMonitor)
+        fakeEventKey =
+            BitmapMemoryCacheKey(
+                fakeCacheEventUri,
+                null,
+                mock(),
+                mock(),
+                null,
+                null,
+                mock()
+            )
+        whenever(mockCacheEvent.cacheKey).thenReturn(fakeEventKey)
+        underTest = DatadogFrescoCacheListener()
+    }
+
+    @AfterEach
+    fun `tear down`() {
+        val isRegistered: AtomicBoolean = GlobalRum::class.java.getStaticValue("isRegistered")
+        isRegistered.set(false)
+    }
+
+    @Test
+    fun `M send RUM error event W read from cache exception`(forge: Forge) {
+        // GIVEN
+        val fakeException: IOException? = forge.aNullable {
+            IOException(forge.aThrowable())
+        }
```

Comment:
I'd feel safer if you had a test with non null throwable and one with null throwable.